### ISSUE TITLE
Custom ordering of challanges

### ISSUE THREAD

**Issue by F0x7r0t** [Hackademic/hackademic#137]
As of now, we have several challenges but we have not yet **categorized** them in anyway. Also the challenges are in random order and is not on the basis of increasing order of difficulty. IMHO, these features should be introduced into hackademic ASAP. For eg:

We can first broadly divide all the challenges to 3 sections mainly Web, crypto and Binary (when we introduce it). Then under each category, challenges should be listed in the order of increasing difficulty (easy - intermediate - Advanced). 

The reason why this is important because students should get some **easy** questions at the beginning so that solving it, will improve their confidence on going ahead (if they solve one question, they will be more motivated to solve next). If we provide challenges in random order and if the student couldn't solve the first question, chances are high that they will quickly **give up**. This should not be the case.

Broad division of challenges could help teachers understand how well the students are doing in each field like web, crypto and binary (this feature should be introduced once we have this issue fixed).

Good to have: it will be good if we have some good **coloring mechanisms** to show different difficulty challenges in different colors (like Red for Advanced).

**Comment by northdpole** [comment]
Good idea, I'd take it a step further and introduce categories to the whole project.
Also give the user the ability to assign categories to articles and challenges and custom order challenges and articles.
That way we can color code challenges depending on the difficulty and propose a categorization of things but if the user wants it, they can change both easily from the front end.

However, we're doing something similar with classes already, so some core changes to support this wouldn't be very hard.

(We'll need to modify the search functionality a bit to add the ability to show categorized items or search within categories and the frontend to color code things and group based on the category)


@pchaigno  what do you think?

(Also, cool new feature idea, a way to extract and import entire courses with challenges and articles)
